### PR TITLE
Switch to using PHP namespaces

### PR DESCRIPTION
**Pull request by joehoyle**
To keep inline with the HM coding standards and consistency with other Human Made code, I'd like to swich S3 Uploads to namespaces, and fix up any outstanding PHP formatting issues with the HM PHPCS.

### CODE REVIEW

**Comment by rmccue** [comment]
#WildIdeas

File: inc/class-plugin.php [hunk 1]
```diff
@@ -257,15 +260,15 @@ public function get_original_upload_dir() : array {
 	 * Reverse a file url in the uploads directory to the params needed for S3.
 	 *
 	 * @param string $url
-	 * @return array
+	 * @return array{bucket: string, key: string, query: string|null}|null
```
Comment (rmccue) [inline inc/class-plugin.php:263]: tihi

**Comment by rmccue** [comment]
I feel like this PR is maybe a bit off-track, do we want to look at shipping this earlier?

**Comment by joehoyle** [comment]
Yeah certainly has gone included a few things -- though the majority is the addition / fixing of type inconsistencies. I think we could still break this up, though a lot of the code-moving issues were brought to light via the type checking and testing fixups.

Either way I think we want to get something merged. This will be for a v3 release, so a breaking change from the last version.

**Comment by joehoyle** [comment]
@rmccue I have create a `v2-branch` off `master` which we can backport things to v2 as needed. By merging this into master, the next major release will be `3.0`, I'd like to get this merged in soon.

**Comment by joehoyle** [comment]
Ping @rmccue for a review on this.

**Comment by joehoyle** [comment]
Ping @rmccue

File: inc/class-plugin.php [hunk 2]
```diff
@@ -157,36 +227,50 @@ public function get_s3_url() {
...
 		if ( empty( $this->original_upload_dir ) ) {
 			wp_upload_dir();
 		}
 
-		return $this->original_upload_dir;
+		/**
+		 * @var array{path: string, basedir: string, baseurl: string, url: string}
+		 */
+		$upload_dir = $this->original_upload_dir;
+		return $upload_dir;
```
Comment (rmccue) [inline inc/class-plugin.php:258]: Why is this a separate var?

File: inc/class-plugin.php [hunk 3]
```diff
@@ -373,19 +461,22 @@ public function is_private_attachment( int $attachment_id ) : bool {
 		 * @param bool Whether the attachment is private.
 		 * @param int  The attachment ID.
 		 */
-		return apply_filters( 's3_uploads_is_attachment_private', false, $attachment_id );
+		$private = apply_filters( 's3_uploads_is_attachment_private', false, $attachment_id );
+		return $private;
```
Comment (rmccue) [inline inc/class-plugin.php:465]: Ditto, why is this a var?

File: inc/class-plugin.php [hunk 4]
```diff
@@ -459,10 +554,12 @@ public function add_s3_signed_params_to_attachment_url( string $url, int $post_i
 		if ( ! $path ) {
 			return $url;
 		}
-		$cmd = $this->s3()->getCommand('GetObject', [
-			'Bucket' => $path['bucket'],
-			'Key' => $path['key'],
-		]);
+		$cmd = $this->s3()->getCommand(
+			'GetObject', [
```
Comment (rmccue) [inline inc/class-plugin.php:558]:
```suggestion
			'GetObject',
			[
```

File: inc/class-stream-wrapper.php [hunk 5]
```diff
@@ -0,0 +1,1210 @@
...
+use Aws\CacheInterface;
+use Aws\LruArrayCache;
+use Aws\Result;
+use Aws\S3\Exception\S3Exception;
+use Aws\S3\S3ClientInterface;
+use Exception;
+use GuzzleHttp\Psr7;
+use GuzzleHttp\Psr7\CachingStream;
+use GuzzleHttp\Psr7\Stream;
+use Psr\Http\Message\StreamInterface;
```
Comment (rmccue) [inline inc/class-stream-wrapper.php:14]: New phpcs error.

File: inc/class-stream-wrapper.php [hunk 6]
```diff
@@ -0,0 +1,1210 @@
...
+		if ( $this->body->isSeekable() ) {
+			$this->body->seek( 0 );
+		}
+		$params = $this->getOptions( true );
+		$params['Body'] = $this->body;
+
+		// Attempt to guess the ContentType of the upload based on the
+		// file extension of the key. Added by Joe Hoyle
+		if ( ! isset( $params['ContentType'] ) &&
+			( $type = Psr7\mimetype_from_filename( $params['Key'] ) )
```
Comment (rmccue) [inline inc/class-stream-wrapper.php:224]: New phpcs error.

File: inc/class-stream-wrapper.php [hunk 7]
```diff
@@ -0,0 +1,1210 @@
...
+				$stat['mode'] = $stat[2] = 0100777;
+				// Pluck the content-length if available.
+				if ( isset( $result['ContentLength'] ) ) {
+					$stat['size'] = $stat[7] = $result['ContentLength'];
+				} elseif ( isset( $result['Size'] ) ) {
+					$stat['size'] = $stat[7] = $result['Size'];
+				}
+				if ( isset( $result['LastModified'] ) ) {
+					// ListObjects or HeadObject result
+					$stat['mtime'] = $stat[9] = $stat['ctime'] = $stat[10]
```
Comment (rmccue) [inline inc/class-stream-wrapper.php:1008]: New-ish phpcs errors.

**Review by rmccue — CHANGES_REQUESTED** [review]
Some minor fixes necessary, but substantially good. I'll review on a tighter timeline next time around :)

File: inc/class-plugin.php (hunk 2 above)
Comment (joehoyle) [reply inc/class-plugin.php:258]: Due to Psalm's comment parsing above an assign expression rather than return.

File: inc/class-plugin.php (hunk 3 above)
Comment (joehoyle) [reply inc/class-plugin.php:465]: Same thing with the php-doc

File: inc/class-stream-wrapper.php (hunk 5 above)
Comment (joehoyle) [reply inc/class-stream-wrapper.php:14]: I'm using it in PHPDOc, thikning maybe that's a PHPCS bug? I can ignore the error though.

File: inc/class-stream-wrapper.php (hunk 6 above)
Comment (joehoyle) [reply inc/class-stream-wrapper.php]: Fixed

File: inc/class-stream-wrapper.php (hunk 5 above)
Comment (joehoyle) [reply inc/class-stream-wrapper.php]: Done

**Comment by joehoyle** [comment]
@rmccue ok ready for re-review

File: inc/class-stream-wrapper.php [hunk 8]
```diff
@@ -10,8 +10,9 @@
 use Exception;
 use GuzzleHttp\Psr7;
 use GuzzleHttp\Psr7\CachingStream;
+use GuzzleHttp\Psr7\MimeType;
 use GuzzleHttp\Psr7\Stream;
-use Psr\Http\Message\StreamInterface;
+use Psr\Http\Message\StreamInterface; //phpcs:ignore Used in Psalm types
```
Comment (rmccue) [inline inc/class-stream-wrapper.php:15]:
```suggestion
use Psr\Http\Message\StreamInterface; //phpcs:ignore -- Used in Psalm types
```

File: inc/class-stream-wrapper.php [hunk 9]
```diff
@@ -10,8 +10,9 @@
 use Exception;
 use GuzzleHttp\Psr7;
```
Comment (rmccue) [inline inc/class-stream-wrapper.php:11]: Looks like this needs an ignore too?
```suggestion
use GuzzleHttp\Psr7; //phpcs:ignore -- Used in Psalm types
```

**Review by rmccue — APPROVED** [review]

**Comment by goldenapples** [comment]
I'm wondering - can we get a new version cut on the v2 branch right before this change was made? This pull request breaks compatability with `altis/cloud`, and I'd like to use a tagged release which includes the Private Attachments functionality from #398 along with altis/cloud v5 or 6.

I think tagging 539d0c1 as 2.3.0 makes sense, and then some of the more recent features can be backported to the v2 branch while `master` is tracking the future v3 branch.

**Comment by joehoyle** [comment]
@goldenapples we can back port any fixes to v2 -- however if private attachments wasn't in v2 (sorry can't remember if it was) I don't think we should back port such a large feature that would be a lot of work.

**Comment by joehoyle** [comment]
@goldenapples it could be that we merged private-attachments before this PR, if that's the case, perhaps we could tag a release, though I'd rather do it as a beta/RC as I don't think we'd want to support that as "working". If your project is already using the `private-attachments` branch instead, it might be best to just back port your fix to that branch.

**Comment by goldenapples** [comment]
No, private attachments wasn't in v2. I can definitely go on using the feature branch (and I've already backported a few fixes to it). Just wanted to check on any release plans and see if it made any sense to do a release containing it.

As long as the `private-attachments` branch doesn't accidentally get cleaned up, I'm fine continuing to track it for now.

**Comment by joehoyle** [comment]
Ok cool -- I'll also open a PR on Altis for the next major version to upgrade to S3 Uploads v3 too

**Pull request merged**